### ISSUE TITLE
Remove deployBaseContracts function and use a flag in contractDeploymentOptions

### ISSUE THREAD

**Issue by Hugoo** [lukso-network/tools-lsp-factory#60]
## I'm submitting a ...
  - feature request
  
## Summary

Remove this function: `lspFactory.DigitalAsset.deployBaseContracts();` which is confusing.

https://docs.lukso.tech/tools/lsp-factoryjs/classes/DigitalAsset#deploybasecontracts

And add a flag in `contractDeploymentOptions` to force deploy from base

```js
lspFactory.DigitalAsset.deployLSP7DigitalAsset(
 { ... },
 { useProxy: false});
```

## TODO:

Delete all following functions:

```
deployBaseContracts
deployLSP7DigitalAsset
deployLSP7DigitalAssetReactive
deployLSP8IdentifiableDigitalAsset
deployLSP8IdentifiableDigitalAssetReactive
```

`reactive: bool` should go in the deployment options, not in the function name.

```
LSP7DigitalAsset.deploy()
LSP8IdentifiableDigitalAsset.deploy()
```

no more `Reactive` or not in the function name.

options:

```
reactive: bool,
```

```
version: '0.1.0' || '0x123456...' (use this as base contract, check if has code) || '0x12345...' (byteCode itself, use this ,no proxy)
```

[] update the docs accordingly
The page explaining the deployment options should be very clear

**Comment by CallumGrindle** [comment]
To confirm how this would work:

`useProxy` = true 
- Uses supplied base contract addresses if provided
- If no base contract addresses supplied, use default base contract addresses stored inside lib
- If no base contract address supplied and default address is empty, deploy new base contracts and deploy UP using new BC
- Return new base contract details if deployed 

`useProxy` = false
- ignore any supplied base contact addresses
- Deploy ‘full contracts’ without proxy deployment

Something to note is that by setting `useProxy` to true, if no valid base contract addresses are passed then new base contracts would be deployed for every new UP. This would be more costly than deploying 'normal' UP contracts

**Comment by CallumGrindle** [comment]
Tasks to be carried out to complete this issue:

- DigitalAsset class -> 2 separate LSP7DigitalAsset + LSP8IdentifiableDigitalAsset
- Allow custom contract bytecode to be passed to deployment functions
- Combine deploy and deployReactive methods
- Remove `deployBaseContracts` methods 
- Add `deployProxy` param to deploy functions. Should default to true

Docs should be updated for all tasks, as development progresses.

These tasks will be addressed this week

**Comment by Hugoo** [comment]
Closed in #69

**Comment by CallumGrindle** [comment]
Closing as this was implemented in #69
